Workspaces/PolyWorkspace/Scripts/PolyGridItem.cs

```diff
@@ -0,0 +1,538 @@
+                yield return null;
+            }
+
+            transform.localScale = targetScale;
+
+            if (callback != null)
+                callback(this);
+
+            m_VisibilityCoroutine = null;
+        }
+
+        // Animate the LocalScale of the asset towards a common/unified scale
+        // used when the asset is magnetized/attached to the proxy, after grabbing it from the asset grid
+        IEnumerator ShowGrabbedObject()
+        {
+            var currentLocalScale = m_DragObject.localScale;
+            var currentPreviewOffset = Vector3.zero;
+            var currentPreviewRotationOffset = Quaternion.identity;
+
+            if (m_PreviewObjectClone)
+                currentPreviewOffset = m_PreviewObjectClone.localPosition;
+
+            var currentTime = 0f;
+            var currentVelocity = 0f;
+            const float kDuration = 1f;
+
+            var targetScale = Vector3.one * k_IconPreviewScale;
+            var pivotOffset = Vector3.zero;
+            var rotationOffset = Quaternion.AngleAxis(30, Vector3.right);
+            if (m_PreviewObjectClone)
+            {
+                var viewerScale = this.GetViewerScale();
+                var maxComponent = m_PreviewBounds.size.MaxComponent() / viewerScale;
+                targetScale = Vector3.one * maxComponent;
+
+                // Object will preview at the same size when grabbed
+                var previewExtents = m_PreviewBounds.extents / viewerScale;
+                pivotOffset = m_PreviewPivotOffset / viewerScale;
+
+                // If bounds are greater than offset, set to bounds
+                if (previewExtents.y > pivotOffset.y)
+                    pivotOffset.y = previewExtents.y;
+
+                if (previewExtents.z > pivotOffset.z)
+                    pivotOffset.z = previewExtents.z;
+
+                if (maxComponent < k_MinPreviewScale)
+                {
+                    // Object will be preview at the minimum scale
+                    targetScale = Vector3.one * k_MinPreviewScale;
+                    pivotOffset = pivotOffset * scaleFactor + (Vector3.up + Vector3.forward) * 0.5f * k_MinPreviewScale;
+                }
+
+                if (maxComponent > k_MaxPreviewScale)
+                {
+                    // Object will be preview at the maximum scale
+                    targetScale = Vector3.one * k_MaxPreviewScale;
+                    pivotOffset = pivotOffset * scaleFactor + (Vector3.up + Vector3.forward) * 0.5f * k_MaxPreviewScale;
+                }
+            }
+
+            while (currentTime < kDuration - 0.05f)
+            {
+                if (m_DragObject == null)
+                    yield break; // Exit coroutine if m_GrabbedObject is destroyed before the loop is finished
+
+                currentTime = MathUtilsExt.SmoothDamp(currentTime, kDuration, ref currentVelocity, 0.5f, Mathf.Infinity, Time.deltaTime);

```

Comment:
Shouldn't this use SmoothInOutLerpFloat instead (so we reach the desired value)

Reply:
Just copied over from AssetGridItem. Will fix.